### ISSUE TITLE
Wrong plots for the data captured through the In-Built MIC

### ISSUE THREAD

**Issue by AsCress** [fossasia/pslab-android#2428]
This issue was also noticed by me in the pre-GSoC period, while writing the proposal.
The plot for the waveform captured using the In-Built MIC doesn't seem to be correct. This can be tested by using a tone generator.
**Actual Behaviour**
The captured waveform has different characteristics (time period and frequency) than the one generated by the tone generator.

**Expected Behaviour**
The captured waveform should have the same characteristics the one generated using the tone generator.

**Steps to reproduce it**
1. Turn on the In-Built MIC feature.
2. Use a tone generator to generate any kind of wave (say, sine) of a particular frequency.
3. You will observe that the one shown in the plot has a very different frequency.

**Screenshots of the issue**
Here, is the output for a sinusoidal waveform of frequency 1000 Hz generated using a tone generator.
![WhatsApp Image 2024-05-23 at 4 36 19 PM](https://github.com/fossasia/pslab-android/assets/125425881/e014c96e-1804-4b64-8a35-c89f4db58c55)
The output waveform has a frequency of approx. 20-25 Hz, which is a huge difference from the input waveform.

**Would you like to work on the issue?**
Yes.

**Comment by AsCress** [comment]
@marcnause Would you be able to confirm this issue, by trying to reproduce it on your devices ?

**Comment by AsCress** [comment]
@CloudyPadmal Is this supposed to work that way? Or, is it an issue ?

**Comment by marcnause** [comment]
> @marcnause Would you be able to confirm this issue, by trying to reproduce it on your devices ?

I tried to reproduce the issue by using the Wave Generator of the PSLab app on one phone and the Oscilloscope screen of the PSLab app on another phone.

The waveform looked similar to the one in the screenshot posted by @AsCress.

I also used the wave generator of the PSLab to generate a sinewave with a frequency of 1000Hz. I connected the Sine Wave 1 output to Channel 1 of the oscilloscope of the PSLab. The result was the waveform in the screenshot below which resembles what I would also expect for the experiment described by @AsCress.

![Screenshot_20240523-165441_094121](https://github.com/fossasia/pslab-android/assets/11596220/c190f69d-0a8b-4a5c-b658-ba2a2ab91931)

I guess this also shows that it is not a general problem of the waveform in the Oscilloscope, but it seems to be limited to the In-Build MIC feature. (I haven't tested the MIC input of the PSLab.)
